### DISCUSSION TITLE
Pushing changes from Weblate to Bitbucket

### DISCUSSION THREAD

**Discussion by perrfect** [WeblateOrg/weblate#9644] [Q&A]
Hello.
I need some help with pushing changes from Weblate to Bitbucket.
I'm using Weblate in docker-compose and trying to configure pushing changes to Bitbucket.
On the Bitbucket repository I've added a public ssh access key from Weblate.
Configured Weblate settings (see screenshot).
![11](https://github.com/WeblateOrg/weblate/assets/39082702/59454806-50bd-4e57-b6af-b8dc8fdcc97a)
But in logs I see and my commits don't appear on Bitbucket:

```
weblate_1   | gunicorn stderr | Acquired Lock('lock:lock:repo:52').                                                        
weblate_1   | gunicorn stderr |  test_company/test-weblate-component/en: committing pending changes (commit)
weblate_1   | gunicorn stderr |  test_company/test-weblate-component/en: committed ['/app/data/vcs/test_company/test-weblate-component/test-weblate-component/en.jso
n'] as Weblate Admin <weblate@example.com>                                                                                                                                                    
weblate_1   | gunicorn stderr | test_company/test-weblate-component: checking triggered alerts                                                               
weblate_1   | gunicorn stderr | test_company/test-weblate-component: skipped push: nothing to push                                                           
weblate_1   | celery-notify stderr |Task weblate.accounts.tasks.notify_change[b125f7c7-0288-4eba-beef-46ff7f2afaa2] received
```

**Comment by nijel** [comment]
When Weblate is configured as "No remote repository" it does not push.

**Comment by perrfect** [comment]
Hello. Thank you for your reply.
Now I'm getting the error:
`Could not push the repo: RepositoryException: fatal: could not read Username for 'https://bitbucket.org': terminal prompts disabled`

I'm using the next settings in a docker-compose:
```
WEBLATE_BITBUCKET_USERNAME=weblate
WEBLATE_BITBUCKET_TOKEN=secret_token
WEBLATE_BITBUCKET_HOST=https://api.bitbucket.org/2.0/repositories/test/test_weblate
```
Is this correct? Or where could be the error?

I added the Bitbucket token to **Repository push URL** field in the web interface directly and it is working.
`https://x-token-auth:some_token_here@bitbucket.org/test/test_weblate.git`
But how can I use bitbucket host, username and token variables in general for all projects?

**Reply by nijel** [reply]
The integration is for creating pull requests and I don't thinki it supports this server. It only works with "BitBucket Server", see https://docs.weblate.org/en/latest/vcs.html#vcs-bitbucket-server

For pushing, it is recommended to use SSH keys and SSH URL, see https://docs.weblate.org/en/latest/vcs.html#ssh-repositories
